### PR TITLE
renesas: g2l: sbc: add option to load and boot CM33 from ATF

### PR DESCRIPTION
**Pull request by vudangRVC**
Support REMOVE_UBOOT option to enable loading and booting CM33 from ATF.
By using that option, U-Boot can be removed from the system boot, and all images for CM33 and CA55 can be loaded and booted from BL2.

### CODE REVIEW

**Comment by vudangRVC** [comment]
Correct the commit title to prevent confusion

File: bl2/bl2_main.c [hunk 1]
```diff
@@ -21,6 +21,11 @@
 
 #include "bl2_private.h"
 
+# ifdef REMOVE_UBOOT
```
Comment (preetam-reddy) [inline bl2/bl2_main.c]: There is a space after '#'. It should be '#ifdef' not '# ifdef'

File: bl2/bl2_main.c [hunk 2]
```diff
@@ -21,6 +21,11 @@
...
+#include <lib/mmio.h>
+#include <drivers/delay_timer.h>
+# endif /* REMOVE_UBOOT */
```
Comment (preetam-reddy) [inline bl2/bl2_main.c]: #endif, no space

File: bl2/bl2_main.c [hunk 3]
```diff
@@ -72,6 +77,107 @@ void bl2_setup(u_register_t arg0, u_register_t arg1, u_register_t arg2,
 }
 #endif /* RESET_TO_BL2 */
 
+# ifdef REMOVE_UBOOT
```
Comment (preetam-reddy) [inline bl2/bl2_main.c]: no space

File: bl2/bl2_main.c [hunk 4]
```diff
@@ -72,6 +77,107 @@ void bl2_setup(u_register_t arg0, u_register_t arg1, u_register_t arg2,
...
+        }
+        else {
+                INFO("BL2: Write register addr = 0x%lx <-- value = 0x%x - passed\n", addr, value);
+        }
+}
+
+void cm33_boot_normal_mode()
+{
+        // Supply clock to CM33_CLKIN
+        write_register(CPG_CLKON_CM33 , 0x00010001);
```
Comment (preetam-reddy) [inline bl2/bl2_main.c]: use macros and avoid hard coding.

File: bl2/bl2_main.c [hunk 5]
```diff
@@ -72,6 +77,107 @@ void bl2_setup(u_register_t arg0, u_register_t arg1, u_register_t arg2,
...
+
+        // Poll CPG_CLKMON_CM33 to confirm that CM33_CLKIN clock is supplied
+        while (mmio_read_32(CPG_CLKMON_CM33) != 0x1)
+                mdelay(10);
+
+        // Stop the reset signals (released from the reset state)
+        write_register(CPG_RST_CM33 , 0x00070007);
```
Comment (preetam-reddy) [inline bl2/bl2_main.c]: use macro and avoid hard coding

File: bl2/bl2_main.c [hunk 6]
```diff
@@ -72,6 +77,107 @@ void bl2_setup(u_register_t arg0, u_register_t arg1, u_register_t arg2,
...
+
+        // Poll CPG_RSTMON_CM33 to confirm that all the reset signals are not applied
+        while(mmio_read_32(CPG_RSTMON_CM33) != 0)
+                mdelay(10);
+}
+
+void cm33_boot_debug_mode()
+{
+        // Supply clock to CM33_TSCLK and CM33_CLKIN
+        write_register(CPG_CLKON_CM33 , 0x00030003);
```
Comment (preetam-reddy) [inline bl2/bl2_main.c]: use macro

File: bl2/bl2_main.c [hunk 7]
```diff
@@ -72,6 +77,107 @@ void bl2_setup(u_register_t arg0, u_register_t arg1, u_register_t arg2,
 }
 #endif /* RESET_TO_BL2 */
 
+# ifdef REMOVE_UBOOT
+/*******************************************************************************
+ * DEFINE CM33
```
Comment (preetam-reddy) [inline bl2/bl2_main.c:82]: Lots of hardcoded values, use macro or const declarations and leave a comment.

File: plat/renesas/rz/common/bl2_plat_mem_params_desc.c [hunk 8]
```diff
@@ -53,20 +57,113 @@ static bl_mem_params_node_t bl2_mem_params_descs[] = {
 		.next_handoff_image_id = BL33_IMAGE_ID,
 	},
 # endif /* BL32_BASE */
+# ifdef REMOVE_UBOOT
```
Comment (preetam-reddy) [inline plat/renesas/rz/common/bl2_plat_mem_params_desc.c]: remove space after #

File: plat/renesas/rz/common/bl2_plat_mem_params_desc.c [hunk 9]
```diff
@@ -53,20 +57,113 @@ static bl_mem_params_node_t bl2_mem_params_descs[] = {
...
+		.image_info.image_max_size = FW_CONFIG_LIMIT - FW_CONFIG_BASE,
+		.image_info.image_base = FW_CONFIG_BASE,
+
+		.next_handoff_image_id = HW_CONFIG_ID,
+	}, /* Finish load for rzv2l_cm33_rpmsg_demo_secure_code.bin */
+	{
+		.image_id = HW_CONFIG_ID,
+
+		SET_STATIC_PARAM_HEAD(ep_info, PARAM_EP, VERSION_2,
+				entry_point_info_t, NON_SECURE | NON_EXECUTABLE),
```
Comment (preetam-reddy) [inline plat/renesas/rz/common/bl2_plat_mem_params_desc.c:81]: NON_SECURE | NON_EXECUTABLE, this is how values should be formed.

File: plat/renesas/rz/common/include/platform_def.h [hunk 10]
```diff
@@ -66,6 +66,46 @@
 #define BL32_LIMIT				(BL32_BASE + 0x100000)
 #endif
```
Comment (preetam-reddy) [inline plat/renesas/rz/common/include/platform_def.h:68]: define the values

File: plat/renesas/rz/common/plat_storage.c [hunk 11]
```diff
@@ -56,7 +56,31 @@ static const io_uuid_spec_t bl32_file_spec = {
 	.uuid = UUID_SECURE_PAYLOAD_BL32,
 };
 
+# ifdef REMOVE_UBOOT
```
Comment (preetam-reddy) [inline plat/renesas/rz/common/plat_storage.c]: remove space after '#'

File: bl2/bl2_main.c (hunk 1 above)
Comment (vudangRVC) [reply bl2/bl2_main.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: bl2/bl2_main.c (hunk 2 above)
Comment (vudangRVC) [reply bl2/bl2_main.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: bl2/bl2_main.c (hunk 3 above)
Comment (vudangRVC) [reply bl2/bl2_main.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: bl2/bl2_main.c (hunk 4 above)
Comment (vudangRVC) [reply bl2/bl2_main.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: bl2/bl2_main.c (hunk 5 above)
Comment (vudangRVC) [reply bl2/bl2_main.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: bl2/bl2_main.c (hunk 6 above)
Comment (vudangRVC) [reply bl2/bl2_main.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: bl2/bl2_main.c (hunk 7 above)
Comment (vudangRVC) [reply bl2/bl2_main.c:82]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: plat/renesas/rz/common/bl2_plat_mem_params_desc.c (hunk 8 above)
Comment (vudangRVC) [reply plat/renesas/rz/common/bl2_plat_mem_params_desc.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

File: plat/renesas/rz/common/include/platform_def.h (hunk 10 above)
Comment (vudangRVC) [reply plat/renesas/rz/common/include/platform_def.h:68]: It's basically the size. No need to define

File: plat/renesas/rz/common/plat_storage.c (hunk 11 above)
Comment (vudangRVC) [reply plat/renesas/rz/common/plat_storage.c]: updated in [ee7f80a](https://github.com/Renesas-SST/rz-atf/commit/ee7f80a0bdfe98d0f2fb27597ac42c71fb9cca68)

**Review by preetam-reddy — APPROVED** [review]

**Pull request merged**